HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,711 @@
+
+Users who wish to *airdrop* (send unsolicited) tokens encounter friction in several ways:
+
+- They must convince users to pre-associate with their token, which is a poor user experience for users who wish to hold the token.
+- If they attempt to airdrop tokens widely, many transactions will simply fail because many accounts have not pre-associated with the token.
+
+Users who wish to receive airdrops:
+
+- Must prepay for auto association slots, which are expensive, even when not filled.
+- Users must hold hbars in order to accept an airdrop (to pay for the association), making it difficult in many workflows to create and pre-populate accounts with tokens.
+
+Retail users from other EVM chains are surprised because other chains do not have the concept of token association. As such, users are accustomed to receiving unsolicited tokens and viewing them in their wallet. Non-Hedera wallets have no native method for understanding token association. Similarly, senders are not used to coordinating token association in advance. This currently provides friction and may serve as a blocker to adoption by developers coming from other chains.
+
+At the same time, many users have no desire to see their wallets flooded with unwanted tokens. For some users the legal and financial implications of these unrequested tokens is undesirable and the management of the tokens is an unwanted burden. An unwanted airdrop is referred to as *spam*.
+
+## Rationale
+
+This HIP improves the user experience by enabling airdrops and enhancing token association workflows on Hedera, while still protecting users and network nodes against security attacks and spam.
+
+### Terminology
+
+- **Airdrop**: Sending an unsolicited token to one or more accounts, often without the account owner having pre-associated with the token.
+- **Sender**: The account sending tokens
+- **Receiver**: The account receiving tokens
+- **Slot**: Each account may define the number of auto-associations it will accept. Each of these is referred to as a “slot”.
+- **Spam**: Unwanted airdrops of tokens
+- **Association renewal credit**: When computing fees for account renewal, a *renewal credit* is worth 1 auto-renew period of payment for an association.
+- **Airdrop-fee**: An additional airdrop specific fee, used to deter indiscriminate spam
+- **Token Airdrop claim**: An additional tranaction type enabling a receiver to obtain tokens in pending state from a sender
+- **Token Airdrop cancel**: An additional tranaction type enabling a sender to remove a previously intended transfer from state
+- **Token reject**: An additional transaction allowing any user to send a token balance back to the token treasury without paying custom fess
```

Comment:
Token Airdrop reject to remain consistent with "claim" and "cancel" ?

Reply:
See https://github.com/hashgraph/hedera-improvement-proposal/pull/904/files#r1507410836
Token Reject doesn't apply to tokens with a pending transfer.
It's added in the HIP to ensure there's a pathway to get rid of tokens you may not want should the need arise.